merlin/io/dataset.py

```diff
@@ -892,14 +893,18 @@ def to_parquet(
                         output_files[fn + suffix] = rgs
             suffix = ""  # Don't add a suffix later - Names already include it
 
+        schema = deepcopy(self.schema)
```

Comment:
I tried `Schema(self.schema.column_schemas)` here, but that' didn't seem to create a sufficiently distinct copy. The dtype changes further down were changing the `self.schema` dtypes as well as the schema copy dtypes

Reply:
I think the reason is probably that `self.schema.column_schemas` is a dictionary, so `Schema(self.schema.column_schemas)` ends up using the same dictionary in both places.